app/controllers/lti_controller.rb

```diff
@@ -67,6 +69,20 @@ def allow_iframe_lti
     response.headers["X-FRAME-OPTIONS"] = "ALLOW-FROM #{session[:lms_domain]}"
   end
 
+  def create_project_if_student_present
+    @user = User.find_by(email: @email_from_lms)
+    @project = Project.find_by(author_id: @user.id, assignment_id: @assignment.id) # find if the project is already present
+    if @project.blank? # if not then create one
+      @project = @user.projects.new
+      @project.name = "#{@user.name}/#{@assignment.name}"
+      @project.assignment_id = @assignment.id
+      @project.project_access_type = "Private"
+      @project.build_project_datum
+      @project.lis_result_sourcedid = params[:lis_result_sourcedid] # this param is required for grade submission
```

Comment:
```suggestion
      @project.lis_result_sourced_id = params[:lis_result_sourced_id] # this param is required for grade submission
```
_ is needed before id to clarify the intention

Reply:
done for ```@projects.``` but for params we need to keep the same ```lis_result_sourcedid``` because it is the default name (as per IMS Global naming) returned by the LMSs